experimental/segmenter/src/provider/lstm.rs

```diff
@@ -0,0 +1,362 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Data provider struct definitions for the lstm
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+
+use crate::math_helper::MatrixZero;
+use icu_provider::prelude::*;
+use zerovec::{ZeroMap, ZeroVec};
+
+/// The struct that stores a LSTM's matrix.
+#[derive(PartialEq, Debug, Clone)]
+pub struct LstmMatrix<'data, const D: usize> {
```

Comment:
Suggestion (optional): use `MatrixZero` itself. We don't really need this wrapper type.

Reply:
I kept them separate for a few reasons: `MatrixZero` is explicitly a borrow, whereas this is borrowed or owned (not sure if that still makes a difference), this expresses dimensions as `u16` for data size (probably irrelevant compared to the matrices though), while MatrixZero converts to `usize` on construction. And, the biggest reason was that I don't want to add too much logic in the provider module.

Reply:
Okay, this makes sense.